XpringKit/Legacy/LegacyDefaultXpringClient.swift

```diff
@@ -52,13 +50,17 @@ extension LegacyDefaultXpringClient: XpringClientDecorator {
   /// - Parameter address: The X-Address to retrieve the balance for.
   /// - Throws: An error if there was a problem communicating with the XRP Ledger or the inputs were invalid.
   /// - Returns: An unsigned integer containing the balance of the address in drops.
-  public func getBalance(for address: Address) throws -> BigUInt {
+  public func getBalance(for address: Address) throws -> UInt64 {
     guard Utils.isValidXAddress(address: address) else {
       throw XRPLedgerError.invalidInputs("Please use the X-Address format. See: https://xrpaddress.info/.")
     }
 
     let accountInfo = try getAccountInfo(for: address)
-    return BigUInt(stringLiteral: accountInfo.balance.drops)
+    guard let drops = UInt64(accountInfo.balance.drops) else {
+      throw XRPLedgerError.malformedResponse("Couldn't parse drops to a balance")
```

Comment:
👍 Nice thinking about edge cases / errors

Reply:
Fun fact, swift uses Optional types. So the type returned from `UInt64(accountInfo.balance.drops)` is actually a Uint64 or nil (in Swift, we'd write this type as `UInt64?`). 

This function defines itself to return a `UInt64` rather than than an optional UInt64 (`UInt64?`) on line 53 so we can't not handle this edge case (it'll be a compiler error). The `guard let` syntax is called optional binding which checks that an optional has a `some` value instead of a `nothing` value.

Reply:
This is very similar to Elm, a "compile-to-JS" language based on FRP and Haskell. Very neat.